source/numerics.tex

```diff
@@ -16034,3 +16034,2871 @@
+
+template<class... Ts>
+  requires @\exposconcept{math-floating-point}@<Ts...>
+    using @\exposid{math-common-simd-t}@ = @\seebelow@;                            // \expos
+
+template<class BinaryOperation, class T>
+  concept @\exposconcept{reduction-binary-operation}@ = @\seebelow@;                    // \expos
+
+// \ref{simd.expos.abi}, \tcode{simd} ABI tags
+template<class T> using @\exposid{native-abi}@ = @\seebelow@;                      // \expos
+template<class T, @\exposid{simd-size-type}@ N> using @\exposid{deduce-abi-t}@ = @\seebelow@;  // \expos
+
+// \ref{simd.flags}, Load and store flags
+struct @\exposid{convert-flag}@;                                                 // \expos
+struct @\exposid{aligned-flag}@;                                                 // \expos
+template<size_t N> struct @\exposid{overaligned-flag}@;                          // \expos
+\end{codeblock}
+
+\rSec3[simd.expos.defn]{Exposition-only helpers}
+
+\begin{itemdecl}
+using @\exposid{simd-size-type}@ = @\seebelow@; // \expos
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\exposid{simd-size-type}{} is an alias for a signed integer type.
+\end{itemdescr}
+
+\begin{itemdecl}
+template<size_t Bytes> using @\exposid{integer-from}@ = @\seebelow@; // \expos
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\tcode{\exposid{integer-from}<Bytes>} is an alias for a signed integer type
+\tcode{T} such that \tcode{sizeof(T)} equals \tcode{Bytes}.
+\end{itemdescr}
+
+\begin{itemdecl}
+template<class T, class Abi>
+  constexpr @\exposid{simd-size-type}\ \exposid{simd-size-v}@ = @\seebelow@; // \expos
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\tcode{\exposid{simd-size-v}<T, Abi>} denotes the width of \tcode{basic_simd<T,
+Abi>} if the specialization \tcode{basic_simd<T, Abi>} is enabled, or \tcode{0}
+otherwise.
+\end{itemdescr}
+
+\begin{itemdecl}
+template<class T> constexpr size_t @\exposid{mask-element-size}@ = @\seebelow@; // \expos
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\tcode{\exposid{mask-element-size}<basic_simd_mask<Bytes, Abi>>} has the value
+\tcode{Bytes}.
+\end{itemdescr}
+
+\begin{itemdecl}
+template<class T> using @\exposid{deduced-simd-t}@ = @\seebelow@; // \expos
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{x} denote an lvalue of type \tcode{const T}.
+
+\pnum
+\tcode{\exposid{deduced-simd-t}<T>} is an alias for
+\begin{itemize}
+ \item
+   \tcode{decltype(x + x)}, if the type of \tcode{x + x} is an enabled
+   specialization of \tcode{basic_simd}; otherwise
+ \item
+   \tcode{void}.
+\end{itemize}
+\end{itemdescr}
+
+\begin{itemdecl}
+template<class V, class T> using @\exposid{make-compatible-simd-t}@ = @\seebelow@; // \expos
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{x} denote an lvalue of type \tcode{const T}.
+\pnum
+\tcode{\exposid{make-compatible-simd-t}<V, T>} is an alias for
+\begin{itemize}
+ \item
+   \tcode{\exposid{deduced-simd-t}<T>}, if that type is not \tcode{void},
+   otherwise
```

Comment:
This was a deliberate choice of LWG because the possible type of the alias follows right after each bullet. I don't care either way, but if the "otherwise" always goes at the start of a bullet (without exceptions) then I think I need to make a few more changes.

Reply:
@tkoeppe , your opinion here?

https://github.com/cplusplus/draft/wiki/Specification-Style-Guidelines isn't very clear on this topic.

Reply:
We seem to have both styles, and I think they're both defensible in context. I don't think we need to mandate a preference at this point.

Reply:
(Obviously stick to a single style within a list.)